### DISCUSSION TITLE
AutoMerge Config Help

### DISCUSSION THREAD

**Discussion by pauldtill** [renovatebot/renovate#16998] [Request Help]
Hi,

I'm looking for some help with automerge, I've reviewed the docs and am not seeing anything which explains my issue.

I have the following in renovate config;
`    {
      groupName: "groupname",
      matchUpdateTypes: [
          "minor",
          "patch",
          "pin",
          "digest"
      ],
      automerge: true,
      matchPackagePatterns: [
          "path1", "path2"
      ],
      semanticCommitType: "ci"
    },`

Renovate picks up changes like this, which is a simple patch update;
![image](https://user-images.githubusercontent.com/27688774/183058217-08f4d888-d3dd-4956-a24c-0c8be35ab7cf.png)

But still shows this in the merge;
![image](https://user-images.githubusercontent.com/27688774/183058343-6f1e2f3d-2d6c-424a-a476-d98cfa1dcdb2.png)

I've enabled debug on renoavate itself and I see this in the output for the specific repo;
`{
    "name": "renovate",
    "branch": "renovate/vault-3.x",
    "msg": "Checking if we can automerge branch",
    "time": "2022-08-05T03:08:22.767Z",
    "v": 0
}
{
    "name": "renovate",
    "branch": "renovate/vault-3.x",
    "msg": "mergeStatus=no automerge",
    "time": "2022-08-05T03:08:22.767Z",
    "v": 0
}`

Why does renovate think this cannot be set for automerge?

Apologies in advance if I've not provided enough info, let me know what else would be needed if so.

Thanks,
Paul

**Comment by rarkins** [comment]
Your package rule isn't matching that dependency. None of the patterns match "vault" from the looks of it

**Reply by pauldtill** [reply]
I think I'd misunderstood the purpose of that pattern list, let me try changing that. Thank you.

**Reply by herndlm** [reply]
quick related question because I also had "Automerge: Disabled by config" problems. For e.g. the package `phpunit/phpunit` being a dev dependency the following should enable automerge, right? Assuming I don't overwrite automerge later again
```json
    {
      "matchDepTypes": ["devDependencies"],
      "matchPackagePatterns": [
        "^@types/",
        "eslint",
        "jest",
        "mockery",
        "phpstan",
        "phpunit"
      ],
      "automerge": true
    },
```

This was working fine with types packages and eslint already. So I recently added a couple more dev deps here, but the last runs did not enable automerge for e.g. phpunit. And I'm trying to figure out why right now

**Reply by herndlm** [reply]
weird, I changed our automerge package rule to the following yesterday and it still shows that it's disabled via config for phpunit MRs
```json
    {
      "description": "Automerge non-major dev updates",
      "matchDepTypes": ["devDependencies"],
      "matchUpdateTypes": ["minor", "patch", "digest", "lockFileMaintenance"],
      "automerge": true
    }
```

time to create a minimal reproducing repo I guess :)

**Reply by herndlm** [reply]
Ok, I figured it out :)
Had to do the following to also work with composer dev deps:
```patch
     },
     {
       "description": "Automerge non-major dev updates",
-      "matchDepTypes": ["devDependencies"],
+      "matchDepTypes": ["devDependencies", "require-dev"],
       "matchUpdateTypes": ["minor", "patch", "digest", "lockFileMaintenance"],
       "automerge": true
     }
```
which makes sense I guess
